src/ProfileEndpoints.jl

```diff
@@ -117,17 +131,25 @@ function _start_cpu_profile(n, delay)
     return resp
 end
 
-function _cpu_profile_response(filename; with_pprof::Bool)
+function handle_cpu_profile_stop(qp, stage_path = nothing)
+    with_pprof = parse(Bool, get(qp, "pprof", default_pprof()))
+    filename = stage_path === nothing ? "cpu_profile" : joinpath(stage_path, "cpu_profile")
+    should_write_data = stage_path !== nothing
+    # Defer the potentially expensive profile symbolication to a non-interactive thread
+    return fetch(Threads.@spawn _cpu_profile_response($filename, $should_write_data; with_pprof=$with_pprof))
+end
+
+function _cpu_profile_response(filename, should_write_data = false; with_pprof::Bool)
     if with_pprof
         prof_name = tempname()
         PProf.pprof(out=prof_name, web=false)
         prof_name = "$prof_name.pb.gz"
-        return _http_response(read(prof_name), "$filename.pb.gz")
+        return _http_create_response_and_maybe_write_data(read(prof_name), "$filename.pb.gz", should_write_data)
```

Comment:
@d-netto I think we can improve this.

As written, this code is now invoking pprof, which exports the data to `prof_name` **on disk**, then reading it back into memory, and then writing it out to disk again, and then sending a link over HTTP.

I think that probably, instead, you should pass the ultimate output path to `pprof` itself, _or_ if for some reason that doesn't work, you could simply issue a `mv()` command to move the file from the current path to the new one. You shouldn't be reading it in and writing it out again.

And in that case: i think that probably it shouldn't be the http response function's responsibility to write anything out at all; that should be handled by the callers, since the details will be specific to each profile type.